KTL.js

```diff
@@ -5620,6 +5622,149 @@ function Ktl($, appInfo) {
             }
         }
 
+        /** _sth = Sticky Table Header
+        * keyword @params numOfRecords, viewHeight - minimum records in table and height of view
+        * @param {string} viewId - The view.key
+        * @param {object} keywords - The keywords object
+        * @param {object} data - records in table*/
+        function stickyTableHeader(viewId, keywords, data) {
+            const kw = '_sth';
+            let numOfRecords = 10;
+            let viewHeight = 800;
+            let isCalledInsideObserver = false;
+
+            if (!keywords[kw]) return;
+
+            if (keywords[kw].length) {
+                numOfRecords = keywords[kw][0].params[0][0] || numOfRecords;
+                viewHeight = keywords[kw][0].params[0][1] || viewHeight;
+            }
+
+            const callStickTableHeaders = () => {
+                if (!isCalledInsideObserver) {
+                    ktl.views.stickTableHeader(viewId, data, numOfRecords, viewHeight);
+                }
+            }
+
+            if (isBulkOpsEnabled(keywords, viewId)) {
+                let observerPromise = new Promise((resolve) => {
+                    observeAndExecute(`#bulkOpsControlsDiv-${viewId}`, () => {
+                        ktl.views.stickTableHeader(viewId, data, numOfRecords, viewHeight);
+                        isCalledInsideObserver = true;
+                        resolve();
+                    });
+                });
+                // If observerPromise does not resolve within 1000ms, call callStickTableHeaders
+                setTimeout(() => {
+                    observerPromise.then(() => {
```

Comment:
What you are doing here is what you think it is. You are only delaying the call to `observerPromise` of 1 sec.
You want to call your promise and then set a timeout.
The question arises: why are you using a promise if you will call `stickTableHeader` a second later anyway? If there is a reason, it would be better to put it in the code, in coding or comments. I firmly believe in descriptive coding (a great read on this is 'Clean Code' by Robert C. Martin).

Also, you should cancel your timeouts instead of using boolean guards. `SetTimeout` returns a unique ID with which you can use `clearTimeout` to cancel its execution.

And, sorry for my long comment, I would suggest leveraging the Promise syntax and writing something like this :
```
let observerPromise = new Promise((resolve) => {
    observeAndExecute(`#bulkOpsControlsDiv-${viewId}`, resolve);
});

observerPromise.then(() => {
        ktl.views.stickTableHeader(viewId, data, numOfRecords, viewHeight);
});
```
It would be easier to read. Fewer JS clutter.

Reply:
Craig, Charles,

When I saw this comment:  `// If observerPromise does not resolve within 1000ms, call callStickTableHeaders`

I had a similar question too:  Why this delay, for what reason, what does it solve as a problem.
It would be nice to explain a bit better in the comment.

Also, I tend to apply Charles recommendations more and more about code simplicity and readability.  I try to make my code almost child-like instead of clever blocks of hard to read/understand code.

I try to use a few extra const/let/var and give them a clear name, then split the code in smaller chunks.

This way, everyone can read it and understand at first glance.

Norm